pkg/cni/cni.go

```diff
@@ -1,15 +1,87 @@
 package cni
 
 import (
+	"encoding/json"
 	"errors"
+	"fmt"
+	"net"
 	"strconv"
 	"strings"
 
+	enov1alpha1 "github.com/Nordix/eno/api/v1alpha1"
+
 	"github.com/Nordix/eno/pkg/cni/cniconfig"
 
 	"github.com/Nordix/eno/pkg/render"
 )
 
+// Ipam is an interface for IPAM Cnis
+type Ipam interface {
+	HandleIpam(cniConf *cniconfig.IpamConfig, d *render.RenderData) error
+}
+
+type WhereAboutsIpam struct{}
+
+type WhereAboutsIpamJson struct {
+	Range  string  `json:"range,omitempty"`
+	Type   string  `json:"type,omitempty"`
+	Routes []Route `json:"routes,omitempty"`
+	Dns    string  `json:"dns,omitempty"`
+}
+
+type Route struct {
+	Destination string `json:"dst,omitempty"`
+	Gateway     string `json:"gw,omitempty"`
+}
+
+//NewWhereAboutsIpam creates instance of WhereAbouts cni
+func NewWhereAboutsIpam() *WhereAboutsIpam {
+	return &WhereAboutsIpam{}
+}
+
+func (ipam *WhereAboutsIpam) HandleIpam(cniConf *cniconfig.IpamConfig, d *render.RenderData) error {
```

Comment:
```suggestion
func (ipam *WhereAboutsIpam) HandleIpam(ipamConf *cniconfig.IpamConfig, d *render.RenderData) error {
```

If we can replicate this change to the whole function would be nice

Reply:
Modified.